lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -120,16 +120,30 @@ def all_values
                     [k.to_sym, vs.first]
                   end.to_h
 
+                  if @values_aggregation_mode == ALL
+                    #  TODO: Can I do this in a less gross way?
+                    pid = /(\d+)\.bin$/.match(file_path)[1]
+                    label_set[:pid] = pid
+                  end
+
                   stores_data[label_set] << v
                 end
               ensure
                 store.close if store
               end
             end
 
-            # Aggregate all the different values for each label_set
-            stores_data.each_with_object({}) do |(label_set, values), acc|
-              acc[label_set] = aggregate_values(values)
+            if @values_aggregation_mode == ALL
+              stores_data.each do |label_set, values|
+                stores_data[label_set] = values.first
+              end
+
+              stores_data
+            else
+              # Aggregate all the different values for each label_set
+              stores_data.each_with_object({}) do |(label_set, values), acc|
+                acc[label_set] = aggregate_values(values)
+              end
```

Comment:
I'd also do this in the `aggregate_values` method below. 

```
           elsif @values_aggregation_mode == ALL
              values.min	              values.first
```

Reply:
I have no idea what I was thinking when I wrote my version. This is so obviously the right way to do it.

Updated!